documentation/specs/static-graph.md

```diff
@@ -227,34 +141,109 @@ We'll represent the project reference protocols as items in MSBuild. The target
+        public string ProjectFile { get; }
+
+        // The global properties to use for this entry point
+        public IDictionary<string, string> GlobalProperties { get; }
+    }
+
+    public class ProjectGraphNode
+    {
+        // No public creation.
+        internal ProjectGraphNode() { }
+
+        // Projects which this project references.
+        IReadOnlyCollection<ProjectGraphNode> ProjectReferences { get; }
+
+        // Projects which reference this project.
+        IReadOnlyCollection<ProjectGraphNode> ReferencingProjects { get; }
+
+        // The evaluated project
+        Project Project { get; }
+    }
+}
+```
 
-**Open Issue:** With this approach initial targets will also get included due to how the RequestBuilder composes entry targets and how TargetBuilder reports build results. Is this an issue? Is it just a perf optimization to take them out from the cache?
+## Isolated builds
+Building a project in isolation means that any build results for project references must be pre-computed and provided as input.
 
-For projects which depend on other projects, the MSBuild task will look up the target result in the cache. If the target is missing from the cache, an error will be logged and the build will fail. If the project is calling into itself either via `CallTarget` or via `MSBuild` with a different set of global properties, this will be always allowed (support for cross-targeting).
+If a project uses the MSBuild task, the build result must be in MSBuild's build result cache instead of just-in-time executing targets on that referenced project. If it is not in the build result cache, an error will be logged and the build will fail. If the project is calling into itself either via `CallTarget` or the MSBuild task with a different set of global properties, this will be allowed to support multitargeting and other build dimensions implemented in a similar way.
 
 Because referenced projects and their entry targets are guaranteed to be in the cache, they will not build again. Therefore we do not need to set `/p:BuildProjectReferences=false` or any other gesture that tells SDKs to not do recursive operations.
 
-Initially this result cache will just be in-memory, similar to the existing MSBuild target result cache, but eventually could persist across builds so that an incremental build of a particular project in the project graph would not require re-evaluation or any target execution in that project's dependencies. This may have significant performance improvements for incremental builds.
+**Open Issue:** With this approach initial targets will also get included due to how the `RequestBuilder` composes entry targets and how `TargetBuilder` reports build results. Is this an issue? Is it just a perf optimization to take them out from the cache? It seems like target mapping incompleteness at the very least, since if the project changes its initial targets, the cache hit from the referencing project would suddenly be a miss.
```

Comment:
This doesn't seem that worrisome but if we ever ship with `InitialTargets` in the cache, we'd have to keep it that way forever. I think that's fine.